spec/support/fixtures/foo

```diff
@@ -438,6 +456,9 @@ Foo::CLI::Commands.register 'hello',       Foo::CLI::Commands::Hello
 Foo::CLI::Commands.register 'greeting',    Foo::CLI::Commands::Greeting
 Foo::CLI::Commands.register 'sub command', Foo::CLI::Commands::Sub::Command
 
+Foo::CLI::Commands.register 'root', Foo::CLI::Commands::Root
+Foo::CLI::Commands.register 'root leaf', Foo::CLI::Commands::Root::Leaf
```

Comment:
I don't believe that anyone will find using classes for namespaces is a good idea. How about `Roots` module? 
```ruby
Foo::CLI::Commands.register 'root', Foo::CLI::Commands::Root
Foo::CLI::Commands.register 'root leaf', Foo::CLI::Commands::Roots::Leaf
```